### ISSUE TITLE
[Bug] LwM2M Transport Service does not support DTLS CID

### ISSUE THREAD

**Issue by Rhyaldir** [thingsboard/thingsboard#10061]
**LwM2M Transport Service does not support DTLS CID**

When a LwM2M client indicates (in the dtls "Client Hello" message) it wants to use the connection_id extension, its request is ignored by the server (the "Server Hello, Server Hello Done" message does not contain the connection_id extension). As a result, a client that has requested the extension and changes ip address or port (for instance due to a longer sleep period) cannot reuse the dtls session. 

Depending on the client implementation this results in:
* a new dtls handshake before sending data (proper implementation) 
* first sending data using the invalid dtls session and only after a timeout starting a new dtls handshake (less proper implementation)
* continue to send data using the invalid dtls session (poor implementation)

For low power devices, even the proper implementation where a new dtls handshake is performed is not desired, as it is a waist of ernergy that can be avoided.

**Server Environment**
* Local docker cluster deployment, build from source
* ThingsBoard Version: 3.6.3-SNAPSHOT
* OS Name and Version: Ubuntu 22.04

**Your Device**
* nRF9160 using custom firmware
* LwM2M v1.1

**To Reproduce**
1. Configure a LwM2M profile and device appropriate for the device
2. Start a network packet dump on the server network interface
3. Let the device connect to the server
4. Notice the Client Hello request contains the "connection_id" extension
5. Notice the Server Hello Done response does not contain the "connection_id" extension
6. Notice packets send by the client from a different port as used by the handhake are not responded to

**Expected behavior**
* The server responds with the "connection_id" extension, with a connection_id set.
* Packets from the client using the given connection_id are accepted and responded to by the server

**Additional context**
The leshan demo server (which uses the same components as used by the Thingsboard LwM2M transport service) does support the dtls CID extension properly.

**Comment by ashvayka** [comment]
Hi @Rhyaldir 

Acknowledged. This bug and the corresponding PR have been sent to our Dev team for review. More details, including the fix version, will be shared soon.

**Comment by Rhyaldir** [comment]
I did some more testing, and found that the current version of the PR alone does not fix the issue completely. 
It does report the CID to the client correctly, but when the client sends some data from a different ip or port it gets a `4.04 Not Found (registration not found) returned`. This is due to the fact that the key of the identity by which it is stored (As implemented by `TblwM2mRedisRegistrationStore`) contains the ip address and port.
This has been long corrected by the leshan team, so I tried upgrading from leshan 2.0.0-M4 to 2.0.0-M14. However as leshan 2 is not stable, a lot of api's were broken; to much to fix easily.
So, I ended up writing a workaround, which seems to do the trick (basically generating a key without the ip address and port when a secure method was used as per  the new leshan code). I will add this new code to the PR tomorrow. In the meantime I will mark it as "WIP".
